File: scope104/src/main/java/com/petro/scope104/presentation/details/UserDetailsActivity.java
package com.petro.scope104.presentation.details;

import android.app.ActivityOptions;
import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.Drawable;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.DataSource;
import com.bumptech.glide.load.engine.GlideException;
import com.bumptech.glide.request.RequestListener;
import com.bumptech.glide.request.target.Target;
import com.petro.scope104.util.IntentHelper;
import com.petro.scope104.R;
import com.petro.scope104.databinding.ActivityUserDetailsBinding;
import com.petro.scope104.domain.entity.WorkerEntity;

import java.text.SimpleDateFormat;
import java.util.Locale;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class UserDetailsActivity extends AppCompatActivity {
    private static final String KEY_WORKER = "KEY_WORKER";
    private ActivityUserDetailsBinding binding;

    public static void start(Context context, WorkerEntity workerEntity, ActivityOptions activityOptions) {
        Intent starter = new Intent(context, UserDetailsActivity.class);
        starter.putExtra(KEY_WORKER, workerEntity);
        context.startActivity(starter, activityOptions.toBundle());
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        final SimpleDateFormat formater = new SimpleDateFormat("dd MMMM yyyy", Locale.getDefault());
        binding = ActivityUserDetailsBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        WorkerEntity workerEntity = ((WorkerEntity) getIntent().getSerializableExtra(KEY_WORKER));

        binding.name.setText(workerEntity.getName());
        binding.phoneNumber.setText(workerEntity.getPhone());
        binding.username1.setText(workerEntity.getUsername());
        binding.email1.setText(workerEntity.getEmail());
        String dateText = String.format(Locale.getDefault(), "%s, %d years", formater.format(workerEntity.getDob()), workerEntity.getAge());
        binding.dob1.setText(dateText);
        binding.nationality1.setText(workerEntity.getNatCountry());
        binding.location1.setText(workerEntity.getCity());
        String dateText1 = formater.format(workerEntity.getRegistered());
        binding.registered.setText(dateText1);
        binding.btnMail.ivIcon.setImageResource(R.drawable.ic_mail);
        binding.btnMail.tvText.setText(R.string.email);
        binding.btnMap.ivIcon.setImageResource(R.drawable.ic_map);
        binding.btnMap.tvText.setText(R.string.map);

        IntentHelper intentHelper = new IntentHelper(this);
        binding.btnPhone.getRoot().setOnClickListener(v -> startActivity(intentHelper.getPhoneIntent(workerEntity.getPhone())));
        binding.btnMail.getRoot().setOnClickListener(v -> startActivity(intentHelper.getEmail(workerEntity.getEmail(), null, null, null)));
        binding.btnMap.getRoot().setOnClickListener(v -> startActivity(intentHelper.getMaps(workerEntity.getCity())));
        binding.profileImage.setTransitionName(getString(R.string.avatarTransition, workerEntity.getUsername()));
        binding.name.setTransitionName(getString(R.string.nameTransition, workerEntity.getUsername()));
        postponeEnterTransition();
        Glide.with(this).load(workerEntity.getAvatarUrlXXL()).circleCrop().addListener(new RequestListener<Drawable>() {
            @Override
            public boolean onLoadFailed(@Nullable GlideException e, Object model, Target<Drawable> target, boolean isFirstResource) {
                startPostponedEnterTransition();
                return false;
            }

            @Override
            public boolean onResourceReady(Drawable resource, Object model, Target<Drawable> target, DataSource dataSource, boolean isFirstResource) {
                startPostponedEnterTransition();
                return false;
            }
        }).into(binding.profileImage);
    }

    @Override
    protected void onDestroy() {
        binding = null;
        super.onDestroy();
    }
}

File: scope104/src/main/kotlin/com/petro/scope104/presentation/details/UserDetailsActivity.kt
package com.petro.scope104.presentation.details

import android.app.ActivityOptions
import android.content.Context
import android.content.Intent
import android.graphics.drawable.Drawable
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import com.bumptech.glide.load.DataSource
import com.bumptech.glide.load.engine.GlideException
import com.bumptech.glide.request.RequestListener
import com.bumptech.glide.request.target.Target
import com.petro.scope104.R
import com.petro.scope104.databinding.ActivityUserDetailsBinding
import com.petro.scope104.domain.entity.WorkerEntity
import com.petro.scope104.util.IntentHelper
import dagger.hilt.android.AndroidEntryPoint
import java.text.SimpleDateFormat
import java.util.*

@AndroidEntryPoint
class UserDetailsActivity : AppCompatActivity() {
    private var binding: ActivityUserDetailsBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val formater = SimpleDateFormat("dd MMMM yyyy", Locale.getDefault())
        binding = ActivityUserDetailsBinding.inflate(layoutInflater)
        setContentView(binding!!.root)
        val workerEntity = intent.getSerializableExtra(KEY_WORKER) as WorkerEntity
        binding!!.name.text = workerEntity.name
        binding!!.phoneNumber.text = workerEntity.phone
        binding!!.username1.text = workerEntity.username
        binding!!.email1.text = workerEntity.email
        val dateText = String.format(
            Locale.getDefault(),
            "%s, %d years",
            formater.format(workerEntity.dob),
            workerEntity.age
        )
        binding!!.dob1.text = dateText
        binding!!.nationality1.text = workerEntity.natCountry
        binding!!.location1.text = workerEntity.city
        val dateText1 = formater.format(workerEntity.registered)
        binding!!.registered.text = dateText1
        binding!!.btnMail.ivIcon.setImageResource(R.drawable.ic_mail)
        binding!!.btnMail.tvText.setText(R.string.email)
        binding!!.btnMap.ivIcon.setImageResource(R.drawable.ic_map)
        binding!!.btnMap.tvText.setText(R.string.map)
        val intentHelper = IntentHelper(this)
        binding!!.btnPhone.root.setOnClickListener { v: View? ->
            startActivity(
                intentHelper.getPhoneIntent(
                    workerEntity.phone
                )
            )
        }
        binding!!.btnMail.root.setOnClickListener { v: View? ->
            startActivity(
                intentHelper.getEmail(
                    workerEntity.email,
                    null,
                    null,
                    null
                )
            )
        }
        binding!!.btnMap.root.setOnClickListener { v: View? ->
            startActivity(
                intentHelper.getMaps(
                    workerEntity.city
                )
            )
        }
        binding!!.profileImage.transitionName =
            getString(R.string.avatarTransition, workerEntity.username)
        binding!!.name.transitionName =
            getString(R.string.nameTransition, workerEntity.username)
        postponeEnterTransition()
        Glide.with(this).load(workerEntity.avatarUrlXXL).circleCrop()
            .addListener(object : RequestListener<Drawable?> {
                override fun onLoadFailed(
                    e: GlideException?,
                    model: Any,
                    target: Target<Drawable?>,
                    isFirstResource: Boolean
                ): Boolean {
                    startPostponedEnterTransition()
                    return false
                }

                override fun onResourceReady(
                    resource: Drawable?,
                    model: Any,
                    target: Target<Drawable?>,
                    dataSource: DataSource,
                    isFirstResource: Boolean
                ): Boolean {
                    startPostponedEnterTransition()
                    return false
                }
            }).into(binding!!.profileImage)
    }

    override fun onDestroy() {
        binding = null
        super.onDestroy()
    }

    companion object {
        private const val KEY_WORKER = "KEY_WORKER"
        fun start(context: Context, workerEntity: WorkerEntity?, activityOptions: ActivityOptions) {
            val starter = Intent(context, UserDetailsActivity::class.java)
            starter.putExtra(KEY_WORKER, workerEntity)
            context.startActivity(starter, activityOptions.toBundle())
        }
    }
}